docs/pages/build-reference/e2e-tests.md

```diff
@@ -243,6 +243,115 @@ If you have set up everything correctly you should see the successful test resul
 
 <img src="/static/images/eas-build/tests/03-logs.png" style={{maxWidth: "100%" }} />
 
+### 6. Upload screenshots of failed test cases
+
+This step is optional but highly recommended. When an E2E test case fails, it can be helpful to see the screenshot of the application state. EAS Build makes it easy to upload any arbitrary build artifacts using the [`buildArtifactPaths`](/build-reference/eas-json.md#buildartifactpaths) field in **eas.json**.
+
+#### Take screenshots for failed tests
+
+[Detox supports taking in-test screenshots of the device](https://wix.github.io/Detox/docs/api/screenshots). It exposes the `device.takeScreenshot()` function that can be called from any test case.
+
+We want to take the screenshot only for failed test cases. Unfortunately, neither `jest` nor `detox` offers a simple way to detect whether a paricular test case has failed. We will implement our own mechanism for that.
+
+Let's start by editing **e2e/environment.js** and adding the `handleTestEvent` method to the `CustomDetoxEnvironment` class. The function handles test events and sets a global `testFailed` variable in the case of test case failure. See the snippet below:
```

Comment:
```suggestion
Neither `jest` nor `detox` offers a simple way to detect when a particular test case has failed and take a screenshot for only failed test cases. You will have to implement your own mechanism for that.

Edit **e2e/environment.js** and add the `handleTestEvent` method to the `CustomDetoxEnvironment` class. The function handles test events and sets a global `testFailed` variable for test case failure. See the snippet below:
```

Let's use [second person narrative](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#write-in-the-second-person) as per our writing style guide.